### ISSUE TITLE
Flaky io.camunda.zeebe.it.gateway.GatewayAuthenticationNoneIT.getTopologyRequestSucceedsWithValidAuthToken

### ISSUE THREAD

**Issue by deepthidevaki** [camunda/camunda#20223]
**Summary**

> Try to answer the following as best as possible

- How often does the test fail?
- Does it block your work?
- Do we suspect that it is a real failure?

**Failures**

```
getTopologyRequestSucceedsWithValidAuthToken(UnaryOperator)[2] apiPicker=rest (Flaky Test): io.camunda.zeebe.it.gateway.GatewayAuthenticationNoneIT
java.lang.AssertionError: 

Expecting
  <HttpZeebeFuture[Failed with the following stack trace:
io.camunda.zeebe.client.api.command.ClientException: Failed to apply credentials to request
	at io.camunda.zeebe.client.impl.http.HttpClient.sendRequest(HttpClient.java:166)
	at io.camunda.zeebe.client.impl.http.HttpClient.get(HttpClient.java:104)
	at io.camunda.zeebe.client.impl.command.TopologyRequestImpl.sendHttpRequest(TopologyRequestImpl.java:114)
	at io.camunda.zeebe.client.impl.command.TopologyRequestImpl.sendRestRequest(TopologyRequestImpl.java:88)
	at io.camunda.zeebe.client.impl.command.TopologyRequestImpl.send(TopologyRequestImpl.java:80)
	at io.camunda.zeebe.it.gateway.GatewayAuthenticationNoneIT.sendRequestAndAssertSuccess(GatewayAuthenticationNoneIT.java:187)
	at io.camunda.zeebe.it.gateway.GatewayAuthenticationNoneIT.getTopologyRequestSucceedsWithValidAuthToken(GatewayAuthenticationNoneIT.java:173)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
```

**Logs**

[Test results for [IT] QA Integration Tests.zip](https://github.com/user-attachments/files/16175847/Test.results.for.IT.QA.Integration.Tests.zip)

**Comment by megglos** [comment]
likely related to https://github.com/camunda/camunda/issues/20136, let's resolve that one first and revisit

**Comment by remcowesterhoud** [comment]
I can't reproduce it anymore locally, so I'm going to assume this was fixed with #20136

**Comment by stephanpelikan** [comment]
I had this right not with `Camunda Spring SDK 8.8-alpha8`. The token stored in `~/.camunda/credentials` was expired but not refreshed automatically. I had to remove the file.
